### PR TITLE
Update Amazon Pay sign-in + UI stub

### PR DESCRIPTION
**Pull request by damiendaemon**
To support the frictionless Alexa checkout Amazon Pay session signing has been integrated into the Retail Demo Store.
To showcase this, we allow users to enter their Amazon Pay details in CloudFormation and get an Amazon Pay button on the Retail Demo Store.
The button is a stub and if the Amazon Pay sandbox details were entered correctly, after starting the session will redirect to the Amazon Pay documentation.
Upcoming work to be delivered the Alexa skill will allow full checkout in the Alexa sandbox.

 -----
 This work was done by the Dae.mn Team. http://dae.mn/ml
 Damien.Duff@dae.mn Joe.Major@dae.mn Kyle.Redelinghuys@dae.mn

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/web-ui/src/partials/AppModal/DemoGuide/DemoGuideArticle/articles/AmazonPayIntegration.vue [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    <ArticleFeature>
+      <template #title>How to Set up Amazon Pay Integration</template>
+      <template #default>
+        <p>
+          In order to set up the integration so that you can initiate a checkout session, we need to set up an
+          Amazon Pay merchant account, as well as a store, and obtain the integration credentials for that store. Having
+          obtained these credentials we can supply them to Retail Demo Store so that it can start a credentialed
+          checkout session which you can enter by authenticating a test user. For more information about integrating
+          with Amazon Pay see <a href="https://developer.amazon.com/docs/amazon-pay-checkout/get-set-up-for-integration.html">
+          these instructions</a>.
```
Comment (james-jory) [inline src/web-ui/src/partials/AppModal/DemoGuide/DemoGuideArticle/articles/AmazonPayIntegration.vue:32]: These step by step instructions in this file sound more appropriate for the repo than in the demo guide. The demo guide is intended for someone receiving a demo and to better understand Amazon Pay and how to experience it in the demo. We're not trying to explain how they'd set up and deploy the RDS with Pay.

File: src/web-ui/src/public/components/AmazonPayButton.vue [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      }
+    },
+    mounted () {
+      this.loadAmazonPayButton()
+    },
+    methods: {
+        async loadAmazonPayButton() {
+            let recaptchaScript = document.createElement('script')
+            recaptchaScript.setAttribute('src', 'https://www.google.com/recaptcha/api.js')
+            document.head.appendChild(recaptchaScript)
```
Comment (james-jory) [inline src/web-ui/src/public/components/AmazonPayButton.vue:34]: What is the purpose of this?

File: src/web-ui/package.json [hunk 3]
```diff
@@ -53,6 +53,9 @@
       "parser": "babel-eslint"
     }
   },
+  "globals": {
+    "amazon": true
+  },
```
Comment (james-jory) [inline src/web-ui/package.json:58]: Why is this needed?

**Review by james-jory — CHANGES_REQUESTED** [review]
Will the Pay signing Lambda be used in other places besides the Carts service? Guessing for the forthcoming Alexa integration but just want to make sure. 

Also, this PR needs #260 merged.

See other detailed comments.

File: src/aws-lambda/amazon-pay-signing/package.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "location-amazon-pay",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "",
+  "license": "ISC",
```
[the remark this answers is not in the document]
Comment (damiendaemon) [reply src/aws-lambda/amazon-pay-signing/package.json]: It was a default. Now removed.

File: src/web-ui/src/public/components/AmazonPayButton.vue (hunk 2 above)
Comment (damiendaemon) [reply src/web-ui/src/public/components/AmazonPayButton.vue]: This has been fixed now. New code:

            if (!document.getElementById("amazon-pay-checkout-javascript")) {
              let payscript = document.createElement('script')
              payscript.setAttribute('src', 'https://static-na.payments-amazon.com/checkout.js')
              payscript.setAttribute('id', 'amazon-pay-checkout-javascript')
              document.head.appendChild(payscript)
            }

File: src/web-ui/package.json (hunk 3 above)
Comment (damiendaemon) [reply src/web-ui/package.json]: It is not. It has been removed.

**Comment by damiendaemon** [comment]
Thank you for the comments. They are very useful.
To answer the first question: the signing Lambda is not used in the forthcoming Alexa changes.
The rest of the comments have been addressed.
To see the changes made:

    git pull
    git diff 9d4975d0 80421c08

**Review by james-jory — CHANGES_REQUESTED** [review]
Looks like #260 changes are not merged yet. Also, the changes to #261 are included in this PR but there are some changes that need to made to that PR before it's ready.

**Comment by damiendaemon** [comment]
> Looks like #260 changes are not merged yet. Also, the changes to #261 are included in this PR but there are some changes that need to made to that PR before it's ready.

Wİth apologies, and thanks for picking this up. It looks like #260 was the only one left out and #261 is now updated with the requested changes.

File: aws/cloudformation-templates/template.yaml [hunk 5]
```diff
@@ -239,6 +254,33 @@ Parameters:
...
+      For more information see "Amazon Pay Integration" under the README file in the amazon-pay-signing subfolder in the repository.
+
+  AmazonPayPrivateKey:
+    Type: String
+    Description: >
+      The private key to be used for signing Amazon Pay requests for a sandbox account.
+      You will need to copy the whole contents of the .pem file into this parameter.
+      For more information see "Amazon Pay Integration" under the README file in the amazon-pay-signing subfolder in the repository.
+    NoEcho: true
+
```
Comment (james-jory) [inline aws/cloudformation-templates/template.yaml:283]: Sorry I missed this earlier. These parameters all need suitable defaults in the workshop template [base-workshop.yaml](https://github.com/aws-samples/retail-demo-store/blob/master/aws/cloudformation-templates/event-engine/base-workshop.yaml) such that the project deploys and operates cleanly without Amazon Pay configured. The workshop template is used to launch 1:many workshop experiences.

**Review by james-jory — CHANGES_REQUESTED** [review]

File: aws/cloudformation-templates/template.yaml (hunk 5 above)
Comment (damiendaemon) [reply aws/cloudformation-templates/template.yaml:283]: OK this is done now, and we have tested as far as initiating deployment with the workshop template. Hopefully next time I will remember to do this before prompting! 

I have also taken the liberty of adding one commit making whether or not Amazon Pay is enabled more declarative in the UI (`Checkout.vue`):

    <amazon-pay-button v-if="placeOrderEnabled && amazonPayEnabled" @click.native="finalizeAmazonPayOrder"/>

...

    amazonPayEnabled() {
      const enabled = process.env.VUE_APP_AMAZON_PAY_PUBLIC_KEY_ID && process.env.VUE_APP_AMAZON_PAY_PUBLIC_KEY_ID !== '' &&
                      process.env.VUE_APP_AMAZON_PAY_STORE_ID && process.env.VUE_APP_AMAZON_PAY_STORE_ID !== '' &&
                      process.env.VUE_APP_AMAZON_PAY_MERCHANT_ID && process.env.VUE_APP_AMAZON_PAY_MERCHANT_ID !== ''
      if(enabled) {
        console.log('Amazon Pay Button Enabled')
      }
      return enabled;
    }

**Review by james-jory — APPROVED** [review]
Looks good to me. Thank you!
